src/sheldon_dictionary.erl

```diff
@@ -0,0 +1,156 @@
+%%% @doc This gen_server only creates the ets for the dictionary and
+%%%      ets for bazingas.
+%%%
+%%% Copyright 2016 Inaka &lt;hello@inaka.net&gt;
+%%%
+%%% Licensed under the Apache License, Version 2.0 (the "License");
+%%% you may not use this file except in compliance with the License.
+%%% You may obtain a copy of the License at
+%%%
+%%%     http://www.apache.org/licenses/LICENSE-2.0
+%%%
+%%% Unless required by applicable law or agreed to in writing, software
+%%% distributed under the License is distributed on an "AS IS" BASIS,
+%%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%%% See the License for the specific language governing permissions and
+%%% limitations under the License.
+%%% @end
+%%% @copyright Inaka <hello@inaka.net>
+%%%
+-module(sheldon_dictionary).
+-author("Felipe Ripoll <ferigis@gmail.com>").
+
+-behaviour(gen_server).
+
+%% API
+-export([
+          start_link/1
+        , exist_in_dictionary/2
+        , dictionary_name/1
+        , get_bazinga/1
+        ]).
+
+%% gen_server callbacks
+-export([
+          init/1
+        , handle_call/3
+        , handle_cast/2
+        , handle_info/2
+        , terminate/2
+        , code_change/3
+        ]).
+
+
+-export_type([
+                language/0
+             ]).
+
+-type language() :: eng.
+
+%%%===================================================================
+%%% API
+%%%===================================================================
+
+-spec(start_link(language()) ->
+  {ok, Pid :: pid()} | ignore | {error, Reason :: term()}).
+start_link(Lang) ->
+  gen_server:start_link({local, ?MODULE}, ?MODULE, [Lang], []).
+
+-spec exist_in_dictionary(language(), string()) -> boolean().
+exist_in_dictionary(Lang, Word) ->
+  DictName = dictionary_name(Lang),
+  LowerWord = string:to_lower(Word),
+  case ets:lookup(DictName, LowerWord) of
+    [] -> false;
+    _  -> true
+  end.
+
+-spec get_bazinga(language()) -> string().
+get_bazinga(Lang) ->
+  BazingaName = bazinga_name(Lang),
+  Keys = ets:tab2list(BazingaName),
+  {Bazinga} = lists:nth(random:uniform(length(Keys)), Keys),
+  Bazinga.
+
+-spec dictionary_name(language()) -> atom().
+dictionary_name(Lang) ->
+  Bin = <<(atom_to_binary(sheldon, utf8))/binary, "_",
+  (atom_to_binary(Lang, utf8))/binary>>,
+  binary_to_atom(Bin, utf8).
+
+%%%===================================================================
+%%% gen_server callbacks
+%%%===================================================================
+
+-spec(init([language()]) -> {ok, State :: map()}).
+init([Lang]) ->
+  ok = learn_language(Lang),
+  ok = set_bazingas(Lang),
+  {ok, #{}}.
+
+-spec(  handle_call(Request :: term()
```

Comment:
Be consistent, if you are using parenthesis here, use it everywhere for each spec.